src/fpm_compiler.f90

```diff
@@ -416,12 +416,79 @@ subroutine set_preprocessor_flags (id, flags)
     do i = 1, size(package%preprocess)
         if (package%preprocess(i)%name == "cpp") then
             flags = flag_cpp_preprocessor// flags
+            call get_macros_from_manifest(id, flags, i)
             exit
         end if
```

Comment:
Should we add:

```fortran
else
    write(stderr, '(a)') 'Warning: preprocessor ' // package%preprocess(i)%name // ' is not supported; will ignore it'
```
or similar? Otherwise the user gets a silent ignore.

More generally, @fortran-lang/fpm, does fpm have a style recommendation on emitting warnings or errors from fpm's internal modules? Is it OK to do it and is there a preferred format?

Reply:
Not yet, I'm hoping to upgrade the manifest reading to the upcoming TOML Fortran 0.3.0 release.

This would allow us to create warnings and errors with rich context information as shown in https://toml-f.readthedocs.io/en/latest/tutorial/fpm-lint.html and https://toml-f.readthedocs.io/en/latest/how-to/error.html.

Reply:
> Should we add:
> 
> ```fortran
> else
>     write(stderr, '(a)') 'Warning: preprocessor ' // package%preprocess(i)%name // ' is not supported; will ignore it'
> ```
> 
> or similar? Otherwise the user gets a silent ignore.
> 

Thanks for noticing this. I fixed this in my new commits.